builder/azure/arm/template_factory.go

```diff
@@ -92,7 +92,10 @@ func GetVirtualMachineDeployment(config *Config) (*resources.Deployment, error)
 			config.ImageSku,
 			config.ImageVersion)
 
-		builder.SetManagedMarketplaceImage(config.Location, config.ImagePublisher, config.ImageOffer, config.ImageSku, config.ImageVersion, imageID, config.managedImageStorageAccountType, config.diskCachingType)
+		err = builder.SetManagedMarketplaceImage(config.Location, config.ImagePublisher, config.ImageOffer, config.ImageSku, config.ImageVersion, imageID, config.managedImageStorageAccountType, config.diskCachingType)
```

Comment:
could this be a breaking change?  I don't know the context of `SetManagedMarketplaceImage` but what happened before when this failed? 🤔

Reply:
https://github.com/hashicorp/packer-plugin-azure/blob/6c11375deca7709accbc00403181db532be99cd7/builder/azure/common/template/template_builder.go#L157

It looks like this method only fails when a specific field isn't set in the template so it might make sense for it to fail here, but I think it's worth changing this back just so we aren't making a potential breaking change (even a logically correct one) in this PR